tools/c7n_oci/c7n_oci/resources/virtual_network.py

```diff
@@ -0,0 +1,387 @@
+# Copyright The Cloud Custodian Authors.
+# SPDX-License-Identifier: Apache-2.0
+
+import logging
+import re  # noqa
+import copy  # noqa
+
+import oci.core
+
+from c7n.filters import Filter, ValueFilter  # noqa
+from c7n.utils import type_schema
+from c7n_oci.actions.base import OCIBaseAction, RemoveTagBaseAction
+from c7n_oci.provider import resources
+from c7n_oci.query import QueryResourceManager
+
+log = logging.getLogger("custodian.oci.resources.virtual_network")
+
+
+@resources.register("cross_connect")
+class Cross_connect(QueryResourceManager):
+    """Oracle Cloud Infrastructure Cross_connect Resource
+
+    :example:
+
+    Returns all Cross_connect resources in the tenancy
+
+    .. code-block:: yaml
+
+        policies:
+            - name: find-all-cross_connect-resources
+              resource: oci.cross_connect
+
+    """
+
+    class resource_type:
+        doc_groups = ["Network"]
+        service = "oci.core"
+        client = "VirtualNetworkClient"
+        enum_spec = ("list_cross_connects", "items[]", None)
+        extra_params = {"compartment_id"}
+        resource_type = "OCI.VirtualNetwork/Cross_connect"
+        id = "id"
+        name = "display_name"
+        search_resource_type = "crossconnect"
+
+
+@Cross_connect.action_registry.register("update-cross-connect")
+class UpdateCrossConnect(OCIBaseAction):
+    """
+    Update cross connect Action
+
+    :example:
+
+    Updates the specified cross-connect.
+
+    Please refer to the Oracle Cloud Infrastructure Python SDK documentation for parameter details to this action
+    https://docs.oracle.com/en-us/iaas/tools/python/latest/api/core/client/oci.core.VirtualNetworkClient.html#oci.core.VirtualNetworkClient.update_cross_connect
+
+    .. code-block:: yaml
+
+        policies:
+            - name: perform-update-cross-connect-action
+              resource: oci.cross_connect
+              actions:
+                - type: update-cross-connect
+
+    """  # noqa
+
+    schema = type_schema(
+        "update-cross-connect", params={"type": "object"}, rinherit=OCIBaseAction.schema
+    )
+
+    def perform_action(self, resource):
+        client = self.manager.get_client()
+        params_dict = {}
+        params_model = {}
+        if self.data.get("params") and self.data.get("params").get("cross_connect_id"):
```

Comment:
I wonder if it would be easier to unconditionally assemble params, and then use c7n.utils.filter_empty to clean it before model construction. that would cut out a lot of boilerplate conditionals.

also noting again we shouldn't statically define resource ids in action configuration.

Reply:
Removed the statically define resource ids in action configuration. Will be available in next push